File: Problems/448. Find All Numbers Disappeared in an Array/solution2.py
class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        
        result = []
        map = {}
        
        for num in nums: 
            map[num] = True
            
        for n in range(1, len(nums) + 1):
            if n not in map:
                result.append(n)
                
        return result
            
# Time Complexity: O(n)
# Space Complexity: O(n)
# Time used: about 15 mins
# Pattern: Arrays & Hashing

File: Problems/205. Isomorphic Strings/solution2.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
 const isIsomorphic = (s, t) => {
  const mapS = {};
  const mapT = {};
  
  for (let i = 0; i < s.length; i++) {
      if ((s[i] in mapS && mapS[s[i]] !== t[i]) || (t[i] in mapT && mapT[t[i]] !== s[i])) {
          return false;
      }
      mapS[s[i]] = t[i];
      mapT[t[i]] = s[i];
  }
  
  return true;
};

// Time complexity; O(n)
// Space complexity: O(n)
// Pattern: Arrays & Hashing
// Time used: 16min (Had to briefly watch NeetCode's explanation of the problem to understand it)